apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-description-links/src/index.ts

```diff
@@ -0,0 +1,156 @@
+				settings[ 'description' ],
+				'https://wordpress.com/support/wordpress-editor/blocks/image-block/'
+			);
+			break;
+
+		case 'core/cover':
+			settings[ 'description' ] = inlineBlockDescriptionLink(
+				settings[ 'description' ],
+				'https://wordpress.com/support/wordpress-editor/blocks/cover-block/'
+			);
+			break;
+
+		case 'core/buttons':
+			settings[ 'description' ] = inlineBlockDescriptionLink(
+				settings[ 'description' ],
+				'https://wordpress.com/support/wordpress-editor/blocks/buttons-block/'
+			);
+			break;
+
+		case 'core/gallery':
+			settings[ 'description' ] = inlineBlockDescriptionLink(
+				settings[ 'description' ],
+				'https://wordpress.com/support/wordpress-editor/blocks/gallery-block/'
+			);
+			break;
+
+		/**
+		 * A8C Blocks
+		 */
+		case 'premium-content/container':
+		case 'premium-content/subscriber-view':
+		case 'premium-content/logged-out-view':
+		case 'premium-content/buttons':
+			settings[ 'description' ] = inlineBlockDescriptionLink(
+				settings[ 'description' ],
+				'https://wordpress.com/support/wordpress-editor/blocks/premium-content-block/'
+			);
+			break;
+
+		case 'a8c/blog-posts':
+			settings[ 'description' ] = inlineBlockDescriptionLink(
+				settings[ 'description' ],
+				'https://wordpress.com/support/wordpress-editor/blocks/blog-posts-block/'
+			);
+			break;
+
+		/**
+		 * Jetpack Blocks
+		 */
+		case 'jetpack/tiled-gallery':
+			settings[ 'description' ] = inlineBlockDescriptionLink(
+				settings[ 'description' ],
+				'https://wordpress.com/support/wordpress-editor/blocks/tiled-gallery-block/'
+			);
+			break;
+
+		case 'jetpack/slideshow':
+			settings[ 'description' ] = inlineBlockDescriptionLink(
+				settings[ 'description' ],
+				'https://wordpress.com/support/wordpress-editor/blocks/slideshow-block/'
+			);
+			break;
+
+		case 'jetpack/subscriptions':
+			settings[ 'description' ] = inlineBlockDescriptionLink(
+				'Allow readers to receive a newsletter with future posts in their inbox. Subscribers can get notifications through email or the Reader app.',
+				'https://wordpress.com/support/wordpress-editor/blocks/subscription-form-block/'
+			);
+			break;
+
+		case 'jetpack/contact-form':
+			settings[ 'description' ] = inlineBlockDescriptionLink(
+				settings[ 'description' ],
+				'https://wordpress.com/support/wordpress-editor/blocks/form-block/'
+			);
+			break;
+
+		case 'jetpack/layout-grid':
+		case 'jetpack/layout-grid-column':
+			settings[ 'description' ] = inlineBlockDescriptionLink(
+				settings[ 'description' ],
+				'https://wordpress.com/support/wordpress-editor/blocks/layout-grid-block/'
+			);
+			break;
+
+		case 'jetpack/mailchimp':
+			settings[ 'description' ] = inlineBlockDescriptionLink(
+				settings[ 'description' ],
+				'https://wordpress.com/support/wordpress-editor/blocks/mailchimp-block/'
+			);
+			break;
+	}
+
+	if ( applyToChildren ) {
+		name = settings[ 'name' ] as string;
```

Comment:
Did we leave this here by accident? Seems like it will have no action after the `switch`.

Reply:
No, this is by design. Prior to the `switch` I check to see if the block has a parent and if truthy I change the name to the parent name to apply this link to all children. This saves me from having to find every child's block name and adding it to this switch. Then at the end, I change the name back to prevent any other issues and return settings. Each case is only `break`ing and the final return of the settings happens at the very end. But I could be wrong, it all was working in my tests unless I missed something.

Reply:
I see. Well, mutating an argument is generally a bad idea and could lead to hard to trace bugs. I wonder if we could just use another variable/constant instead of mutating the argument itself.

Reply:
Thanks for the insight! I did not think about that at all. I replaced this with a simpler and better solution that eliminates the mutation and it is working. Let me know what you think.

Reply:
Haha I forgot I pushed this... I will make a PR to fix this.

Reply:
No rush at all. Thanks for the quick handling 🙌